src/components/SwapForm/SwapActionButton/index.tsx

```diff
@@ -136,10 +136,10 @@ const SwapActionButton: React.FC<Props> = ({
   const toggleApprovalModal = useToggleModal(ApplicationModal.SWAP_APPROVAL)
 
   const handleApproveClick = () => {
-    if (walletKey && ['METAMASK', 'TRUST_WALLET'].includes(walletKey?.toString())) {
-      approveCallback()
-    } else {
+    if (['COIN98', 'BRAVE', 'COINBASE'].includes(walletKey)) {
```

Comment:
how about these other wallets which dont have custom allowance like metamask?  we should show approval modal

Reply:
new added wallets don't support too